homeassistant/components/ccm15/climate.py

```diff
@@ -0,0 +1,160 @@
+"""Climate device for CCM15 coordinator."""
+import logging
+from typing import Any
+
+from ccm15 import CCM15DeviceState
+
+from homeassistant.components.climate import (
+    FAN_AUTO,
+    FAN_HIGH,
+    FAN_LOW,
+    FAN_MEDIUM,
+    PRECISION_WHOLE,
+    SWING_OFF,
+    SWING_ON,
+    ClimateEntity,
+    ClimateEntityFeature,
+    HVACMode,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_TEMPERATURE, UnitOfTemperature
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import CONST_CMD_FAN_MAP, CONST_CMD_STATE_MAP, DOMAIN
+from .coordinator import CCM15Coordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class CCM15Climate(CoordinatorEntity[CCM15Coordinator], ClimateEntity):
+    """Climate device for CCM15 coordinator."""
+
+    _attr_should_poll = True
+    _attr_temperature_unit = UnitOfTemperature.CELSIUS
+    _attr_has_entity_name = True
+    _attr_target_temperature_step = PRECISION_WHOLE
+    _attr_hvac_modes = [
+        HVACMode.OFF,
+        HVACMode.HEAT,
+        HVACMode.COOL,
+        HVACMode.DRY,
+        HVACMode.AUTO,
+    ]
+    _attr_fan_modes = [FAN_AUTO, FAN_LOW, FAN_MEDIUM, FAN_HIGH]
+    _attr_swing_modes = [SWING_OFF, SWING_ON]
+    _attr_supported_features = (
+        ClimateEntityFeature.TARGET_TEMPERATURE
+        | ClimateEntityFeature.FAN_MODE
+        | ClimateEntityFeature.SWING_MODE
+    )
+
+    def __init__(
+        self, ac_host: str, ac_index: int, coordinator: CCM15Coordinator
+    ) -> None:
+        """Create a climate device managed from a coordinator."""
+        super().__init__(coordinator)
+        self._ac_host: str = ac_host
+        self._ac_index: int = ac_index
+        self._attr_name = str(self._ac_index)
+        self._attr_unique_id = f"{self._ac_host}.{self._ac_index}"
+        self._attr_device_info = DeviceInfo(
+            identifiers={
+                # Serial numbers are unique identifiers within a specific domain
+                (DOMAIN, f"{self._ac_host}.{self._ac_index}"),
+            },
+            name=self._attr_name,
+            manufacturer="Midea",
+            model="CCM15",
+        )
```

Comment:
```suggestion
    )
    _attr_name = None

    def __init__(
        self, ac_host: str, ac_index: int, coordinator: CCM15Coordinator
    ) -> None:
        """Create a climate device managed from a coordinator."""
        super().__init__(coordinator)
        self._ac_host: str = ac_host
        self._ac_index: int = ac_index
        self._attr_unique_id = f"{self._ac_host}.{self._ac_index}"
        self._attr_device_info = DeviceInfo(
            identifiers={
                # Serial numbers are unique identifiers within a specific domain
                (DOMAIN, f"{self._ac_host}.{self._ac_index}"),
            },
            name=str(self._ac_index),
            manufacturer="Midea",
            model="CCM15",
        )
```

Reply:
setting `_attr_name` to `None` makes it use the device name as entity name.

Reply:
This will make the devices harder to indentify, ccm15 does not provide a way to get firnedly names to the slaves, only and index:
![image](https://github.com/home-assistant/core/assets/2091582/c30c731e-f8a9-4902-b13e-1e82bd53ac67)

Reply:
It's literally the same. Did you see I also change the device name to `str(self._ac_index)`?